### PR TITLE
Documentation: Add integration & production user tracking docs

### PR DESCRIPTION
**Pull request by aaronlevy**
I'd still like to gather a few more use-cases before initial merging. This is to get the scaffolding started.

Also on the table is what info we want to collect RE production users (can use etcd as a reference: https://github.com/coreos/etcd/blob/master/Documentation/production-users.md)

### CODE REVIEW

**Comment by philips** [comment]
LGTM!

File: Documentation/users-integrations.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Users and Integrations
+
+This document tracks projects, integrations, and use cases for bootkube. [Join the community](../README.md), and help us keep the list up-to-date.
+
+* [Tectonic](https://coreos.com/tectonic) - Enterprise-ready Kubernetes clusters
```
Comment (xiang90) [inline Documentation/users-integrations.md]: add https://github.com/poseidon/bootkube-terraform ?

Comment (xiang90) [inline Documentation/users-integrations.md]: https://github.com/kahkhang/kube-linode

Comment (dghubble) [inline Documentation/users-integrations.md]: https://github.com/poseidon/typhoon - Minimal and free Kubernetes distribution
https://github.com/poseidon/bootkube-terraform - Terraform module for rendering bootkube manifests and assets

Comment (aaronlevy) [inline Documentation/users-integrations.md]: Added all 3, thanks!

**Pull request merged**